sklearn/tree/export.py

```diff
@@ -412,6 +416,12 @@ def recurse(tree, node_id, criterion, parent=None, depth=0):
                                "; ".join(r for r in ranks[rank]) + "} ;\n")
         out_file.write("}")
 
+        # If out_file is a StringIO buffer, return its contents
```

Comment:
This may change some former code that was expecting no return value... It might be better to more explicitly have a flag for "out_file_none".